### PR TITLE
feat: replace react-select-plus with Combobox

### PR DESCRIPTION
**Pull request by mcasimir**
Replaces `react-select-plus` with the `Combobox` component:

https://user-images.githubusercontent.com/334881/198715479-dd7db2de-8db2-427d-9361-3d4066cb255a.mp4

https://user-images.githubusercontent.com/334881/198715985-578eba22-a272-4ebd-8633-d31280ebd657.mp4

I've accidentally replaced the select on the table view as well as initially I thought that was the library that is not compatible with react 17 

Screenshot 2022-10-28 at 21 07 01

### CODE REVIEW

File: packages/compass/src/app/styles/deprecated-bootstrap-styles/globals.less [hunk 1]
```diff
@@ -21,7 +21,7 @@
   .checkbox input[type="checkbox"] {
     position: absolute;
     margin-left: -20px;
-    margin-top: 4px \\9;
+    margin-top: 4px;
```
Comment (mcasimir) [inline packages/compass/src/app/styles/deprecated-bootstrap-styles/globals.less:24]: drive-by fix

before:
Screenshot 2022-10-28 at 20 53 44

after:
Screenshot 2022-10-28 at 20 53 14

File: packages/compass-export-to-language/package.json [hunk 2]
```diff
@@ -54,6 +54,7 @@
     "reformat": "npm run prettier -- --write ."
   },
   "peerDependencies": {
+    "@mongodb-js/compass-components": "*",
```
Comment (gribnoysup) [inline packages/compass-export-to-language/package.json:57]: Star versions lead to weird behavior with npm@8, I cleaned it up awhile ago, but seems like this is getting introduced again. Can you clean these up please in this package while you are adding new dependencies here?

Comment (mcasimir) [reply packages/compass-export-to-language/package.json:57]: Sure!

Comment (Anemy) [reply packages/compass-export-to-language/package.json:57]: If we find this happens often it may be worth adding something to our `depalign` script.

File: packages/compass/src/main/window-manager.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ const DEFAULT_HEIGHT = (() => {
   return height;
 })();
 
-const MIN_WIDTH = process.env.COMPASS_MIN_WIDTH ?? 1024;
+const MIN_WIDTH = process.env.COMPASS_MIN_WIDTH ?? 1025;
```
Comment (Anemy) [inline packages/compass/src/main/window-manager.ts:46]: How do you feel about leaving a comment about why we have the min width set at that amount?
```suggestion
// We set the min width to 1025 so that the screensize breakpoints of leafygreen
// components are not hit. The breakpoints make the styles of the Select component
// change significantly at widths of 1024 and less.
const MIN_WIDTH = process.env.COMPASS_MIN_WIDTH ?? 1025;
```

File: packages/compass/src/app/styles/index.less [hunk 4]
```diff
@@ -12,7 +12,6 @@
 @import './deprecated-bootstrap-styles/grid.less';
 
 // Dependencies
-@import './react-select.less';
```
Comment (Anemy) [inline packages/compass/src/app/styles/index.less:15]: Can we also remove this file now? Or are we going to clean up all of the files later on?

File: packages/compass-crud/src/components/table-view/types-dropdown.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import TypeChecker from 'hadron-type-checker';
+import { Select, Option, css, spacing } from '@mongodb-js/compass-components';
+
+const CASTABLE_TYPES = TypeChecker.castableTypes(true);
+
+const selectStyles = css({ minWidth: spacing[3] * 10 });
+
+const TypesDropdown = ({ element }) => {
```
Comment (Anemy) [inline packages/compass-crud/src/components/table-view/types-dropdown.jsx:10]: Is there a reason we don't want this file in typescript? `compass-crud` is on the new config. We might be able to use some of the types from `hadron-document` as well.  Same q with the `types-dropdown.spec.jsx`

File: packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx [hunk 6]
```diff
@@ -1,61 +1,82 @@
...
-          value={this.props.selectedStage}
-          onChange={this.onStageOperatorSelected}
-        />
-      </div>
-    );
-  }
+function comboboxOptionStyles(stage) {
+  return css({
+    '&::after': {
+      content: JSON.stringify(
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx:26]: Just a thought, it might be worth leaving a comment why we show the combobox description inside of css. Nice workaround btw!

**Review by Anemy — APPROVED** [review]
lgtm! A couple ideas/suggestions, no blockers.

File: packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx [hunk 7]
```diff
@@ -1,61 +1,82 @@
...
+}) => {
+  const onStageOperatorSelected = useCallback((name) => {
+    onChange(index, name);
+  }, [onChange, index]);
+
+  const optionStyleByStageName = useMemo(() => {
+    return Object.fromEntries(stages.map((stage) => [stage.name, comboboxOptionStyles(stage)]))
+  }, [stages])
+
+  return <Combobox value={selectedStage}
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx:55]: I think it's fine as is, however if we want to make this scroll a bit nicer with the aggregations content we could pass a `scrollContainer` reference to the aggregation pipeline window.
https://www.mongodb.design/component/combobox/documentation/
